src/gui/plugins/tape_measure/TapeMeasure.qml

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Window 2.2
+import QtQuick.Controls 2.1
+import QtQuick.Controls.Material 2.2
+import QtQuick.Controls.Material.impl 2.2
+import QtQuick.Layouts 1.3
+import QtQuick.Controls.Styles 1.4
+import "qrc:/qml"
+
+ToolBar {
+  id: tapeMeasure
+  Layout.minimumWidth: 200
+  Layout.minimumHeight: 100
+
+  property color snapTitle: (Material.theme == Material.Light) ?
+    Material.color(Material.Grey, Material.Shade900) :
+    Material.color(Material.Grey, Material.Shade200)
+
+  property color snapItem: (Material.theme == Material.Light) ?
+    Material.color(Material.Grey, Material.Shade800) :
+    Material.color(Material.Grey, Material.Shade100)
+
+  property var distance: 0.0
+
+  function updateDistance() {
+    distance = TapeMeasure.Distance();
+  }
+
+  function startMeasurement() {
+    TapeMeasure.OnReset();
+    TapeMeasure.OnMeasure();
+  }
+
+  Connections {
+    target: TapeMeasure
+    onNewDistance: {
+      updateDistance();
+    }
+  }
+
+  background: Rectangle {
+    color: "transparent"
+  }
+
+  ButtonGroup {
+    id: group
+  }
+
+  Shortcut {
+    sequence: "M"
```

Comment:
Can we also use `Esc` to cancel a measurement?

Reply:
I'll create another PR since Shortcuts are unique across an application for each key event and the `Transform Control` already makes use of `Esc`

Reply:
Added the logic for escape and `m` key press to the cpp side and removed the `Shortcut`  65d122ab3dec6a2f1b23c5c8068501c3eb277129